lib/housing-available/housing-available.controllers.js

```
@@ -1 +1,229 @@
+          try {
+            await Log.request({
+              type: 'property.submit.failed.county',
+              severity: 'Notice',
+              request: request,
+            });
+          } catch (e) { }
+          throw Boom.badData('Sorry, We are only accepting properties in ' + `${process.env.ADDRESS_LIMIT_COUNTY}, ${process.env.ADDRESS_LIMIT_STATE}`);
+        }
+        const addressHash = postalAddressLib.hashAddress(addressObject);
+
+        addressObject.hash = addressHash;
+
+        const existingAddresses = await postalAddressLib.getAddressesByHash(addressHash);
+        if (existingAddresses.length > 0) {
+          housingAvailableObject.PropertyId = existingAddresses[0].PropertyId;
+        } else {
+          // TODO: create new Property and attach it to this;
+        }
+        // let locationTerms = [addressObject.addressRegion];
+        let locationTerms = [addressObject.addressRegion, addressObject.addressLocality];
+        if (addressObject.hasOwnProperty('addressNeighborhood')) {
+          locationTerms.push(addressObject.addressNeighborhood);
+        } else {
+          locationTerms.push(addressObject.addressLocality);
+        }
+        let locationId = await locationLib.locationFindCreateNestedTerms(locationTerms);
+        housingAvailableObject.LocationId = locationId;
+      }
+
+      if (hasAddress) {
+        housingAvailableObject.address = request.payload.address;
+      }
+
+      if (request.payload.contact) {
+        housingAvailableObject.contact = request.payload.contact;
+      }
+
+      if (request.payload.metadata) {
+        housingAvailableObject.metadata = request.payload.metadata;
+      }
+
+      // Create a 'hapi' instance: our hAObject + metadata about it;
+      let housingAvailableInstance;
+      try {
+        housingAvailableInstance = await createHousingAvailable(housingAvailableObject);
+        return h
+          .response(housingAvailableInstance);
+      } catch (e) {
+        return e;
```

we never want to just return an error. Some errors are useful to a user, but some aren't and shouldn't make their way back. If our create fails I think we want to throw a Boom.badImplementation. and you can return the h.response(housingAvailableInstance); after the try..catch block.